src/wp-includes/link-template.php

```diff
@@ -1597,16 +1597,23 @@ function get_delete_post_link( $post = 0, $deprecated = '', $force_delete = fals
  * @since 2.3.0
  *
  * @param int|WP_Comment $comment_id Optional. Comment ID or WP_Comment object.
+ * @param string         $context    Optional. Could be 'display' or 'null'. Default null.
```

Comment:
The default needs to be `display`, similar to how it is for the `get_edit_post_link()` parameter of the same name. `display` should result in the same behavior as currently is present.

The other possible value should be a different string instead of `null`. Looking at `get_edit_post_link()`, while it's undefined at the function implementation, WordPress core uses `url`. So I suggest we do it the same way, the possible values should be `url` and `display` (default).